engine/table/src/main/java/io/deephaven/engine/table/impl/select/python/ArgumentsChunked.java

```diff
@@ -3,125 +3,105 @@
+import io.deephaven.engine.util.PyCallableWrapper.ConstantChunkArgument;
+import org.jpy.PyObject;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public class ArgumentsChunked {
+    private final Collection<ChunkArgument> chunkArguments;
+    private final Class<?> returnType;
+    private final boolean forNumba;
+
+    private void prepareOneChunkedArg(Object[] chunkedArgs, Class<?>[] chunkedArgTypes, int argIdx, Class<?> argType,
+            Object argValue, int maxChunkSize) {
+        if (argType == byte.class) {
+            chunkedArgTypes[argIdx] = byte[].class;
+            chunkedArgs[argIdx] = new byte[maxChunkSize];
+            if (argValue != null) {
+                Arrays.fill((byte[]) chunkedArgs[argIdx], (byte) argValue);
+            }
+        } else if (argType == boolean.class) {
+            chunkedArgTypes[argIdx] = boolean[].class;
+            chunkedArgs[argIdx] = new boolean[maxChunkSize];
+            if (argValue != null) {
+                Arrays.fill((boolean[]) chunkedArgs[argIdx], (boolean) argValue);
+            }
+        } else if (argType == char.class) {
+            chunkedArgTypes[argIdx] = char[].class;
+            chunkedArgs[argIdx] = new char[maxChunkSize];
+            if (argValue != null) {
+                Arrays.fill((char[]) chunkedArgs[argIdx], (char) argValue);
+            }
+        } else if (argType == short.class) {
+            chunkedArgTypes[argIdx] = short[].class;
+            chunkedArgs[argIdx] = new short[maxChunkSize];
+            if (argValue != null) {
+                Arrays.fill((short[]) chunkedArgs[argIdx], (short) argValue);
+            }
+        } else if (argType == int.class) {
+            chunkedArgTypes[argIdx] = int[].class;
+            chunkedArgs[argIdx] = new int[maxChunkSize];
+            if (argValue != null) {
+                Arrays.fill((int[]) chunkedArgs[argIdx], (int) argValue);
+            }
+        } else if (argType == long.class) {
+            chunkedArgTypes[argIdx] = long[].class;
+            chunkedArgs[argIdx] = new long[maxChunkSize];
+            if (argValue != null) {
+                Arrays.fill((long[]) chunkedArgs[argIdx], (long) argValue);
+            }
+        } else if (argType == float.class) {
+            chunkedArgTypes[argIdx] = float[].class;
+            chunkedArgs[argIdx] = new float[maxChunkSize];
+            if (argValue != null) {
+                Arrays.fill((float[]) chunkedArgs[argIdx], (float) argValue);
+            }
+        } else if (argType == double.class) {
+            chunkedArgTypes[argIdx] = double[].class;
+            chunkedArgs[argIdx] = new double[maxChunkSize];
+            if (argValue != null) {
+                Arrays.fill((double[]) chunkedArgs[argIdx], (double) argValue);
+            }
+        } else if (argType == PyObject.class) {
+            chunkedArgTypes[argIdx] = PyObject[].class;
+            chunkedArgs[argIdx] = new PyObject[maxChunkSize];
```

Comment:
Do we need an explicit string block, or is that handled under PyObject?

Reply:
It is actually handled under the generic Object case.